src/VisualStudio/Core/Def/Implementation/VirtualMemoryNotificationListener.cs

```diff
@@ -61,6 +68,17 @@ public int OnBroadcastMessage(uint msg, IntPtr wParam, IntPtr lParam)
                         }
 
                         _workspaceCacheService.FlushCaches();
+
+                        // turn off full solution analysis
+                        if (_workspace.Options.GetOption(RuntimeOptions.FullSolutionAnalysis))
+                        {
+                            _workspace.Services.GetService<IOptionService>().SetOptions(_workspace.Options.WithChangedOption(RuntimeOptions.FullSolutionAnalysis, false));
+
+                            // let user know full analysis is turned off due to memory concern
+                            // no close info bar action
+                            _workspace.Services.GetService<IErrorReportingService>().ShowErrorInfo(ServicesVSResources.FullSolutionAnalysisOff, () => { });
```

Comment:
why not pass in a callback for 'onEnableClicked' that sets the option back?

THen you also don't need the wonky text about going into "tools options" or needing to restart VS.  The ui would then just be:

"Full solution analysis disabled due to low memory conditions." With an 'enable' button if the user wants to turn it back on right there in the UI.